src/platform/plugins/shared/unified_search/public/search_bar/create_search_bar.tsx

```diff
@@ -253,6 +253,7 @@ export function createSearchBar({
             }
             savedQuery={savedQuery}
             onQuerySubmit={defaultOnQuerySubmit(props, data.query, query)}
+            onQueryChange={props.onQueryChange}
```

Comment:
that's the one change we'd need so we get access to the query that was not submitted in Discover to add it to our state (use it elsewhere)